### DISCUSSION TITLE
Analog output of the current position of the x-axis connected to a attiny 13

### DISCUSSION THREAD

**Discussion by emimic94** [fra589/grbl-Mega-5X#366] [Q&A]
Hi community, 

More or less simple question, I'd like to output an analog signal from the arduino mega. The signal should map the current position of the x carriage. The resolution requirement is quite low, thus absolute distance to machine zero dived by 64 should be sufficient.
Would it be possible to use A14 or A13 for this purpose? 
If yes where to put the code for that operation? and how to access the register where the positions are stored? 

Our aim... we want to add a live camera feature allowing us to live stream and community control the camera. In auto mode the camera should follow the gantry automatically (therefore the position feedback is required). In community control mode the camera gantry could be moved in x and theta axis to change the perspective. 

Thank you! 
Michael

**Comment by fra589** [comment]
Hi @emimic94,  
  
There is no "analogic" output on Mega 2560. To output "variable voltage", it's nedeed to use PWM output.
Considering available timers left by Grbl need, the only pin available for PWM is D9.  
There is an option in config.h (`#defne USE_OUTPUT_PWM`) which permit the use of this PWM output with the M67 & M68 Gcode commands.  
But to have real time output PWM according to axes movments, you need to implement it in the timer interrupt of stepper driver in the stepper.c file. This may have a significant impact on speed performance.  
Why not using the `?` native output of Grbl?  
It might be easier to work in report.c to add output to another serial port.  

@++;
Gauthier.

**Reply by emimic94** [reply]
Hi Gauthier,
thank you for your fast feedback! 
Yes sure, i had the Arduino due in mind which has true DAC output pins... 
Ok the drop in speed performance is an issue so the one wire PWM is no way to go for me. 
Two wire serial communication should be fine too, I could use software uart on the attiny13.

Regarding the code part ... 
1) Add 2nd xserial_write void in serial.c and add the required lines for UBRR1x
2) Add 2nd xprintFloat_CoordValue in print.c 
3) Add 2nd xprintFloat in print.c 
4) Call 2nd xPrintFloat_CoordValue, line prior to n_report++; 

Or is there an easier way to achieve this? 

Thank you in advance! 
Michael
